### DISCUSSION TITLE
Don't treat the post like a wall for collisions / pathfinding

### DISCUSSION THREAD

**Discussion by fidwell** [OpenRCT2/OpenRCT2#17594] [Feature requests]
The vertical post is a great and versatile scenery piece. However, it has one major flaw: the game treats it like a wall. While this is handy for placement, the post does not _appear_ to be a wall, and only takes up a couple of pixels in the corner of the tile. Despite this, it still blocks the entire length of a tile. This means that if you place a post and then place path around it, the path will not connect, even though it looks like it ought to.

I have a hypothesis that this behavior sometimes causes issues with queues breaking when there are posts on them, but can't confirm anything.

![image](https://user-images.githubusercontent.com/5436387/180587928-a2bec086-96e1-4248-a046-9eb1eba315f7.png)

Fig 1: a) A post. b) A path with post placed on top, with clearance checks disabled. c) A post with path placed around it, irrespective of clearance checks. (Undesired outcome.)

**Comment by ocalhoun6** [comment]
There's not a really easy solution to this, though, other than simply disabling clearance checks in situations that could make it problematic.

You could replace the posts with a new scenery object ... but then it would have to be a wall, a quarter-tile object, or a full-tile object, and that would just keep the same problem or make it even worse.

To *really* fix this, you'd have to make a whole new class of object, a 'corner object' or something. With new code to handle the placement of it, new code to handle how it's stored in save files, etc.

Maybe there could be some kind of hard-coded exception in the path-placing code (and probably track-placing code as well) that tells it to ignore walls if the ID of the scenery object matches this one? But that could be kind of ugly code...

Probably best to just keep it as a wall and disable clearance checks when necessary.

**Reply by fidwell** [reply]
Yeah ideally turning it into a "corner object" would be the best solution, but a flag to let it ignore paths would be nice too. There are other corner-style objects that don't behave as walls but fulfill similar functions, so it doesn't necessarily _have_ to be a one-off unique solution.

**Comment by 733737** [comment]
the devs could change it in the next objects release, I don't think it would be too hard
